native/cocos/renderer/core/PassUtils.cpp

```diff
@@ -262,4 +262,45 @@ bool overrideMacros(MacroRecord &target, const MacroRecord &source) {
     return isDifferent;
 }
 
+MaterialProperty toMaterialProperty(gfx::Type type, const ccstd::vector<float> &vec) {
```

Comment:
why change this?

Reply:
Because the logic of https://github.com/cocos/cocos-engine/blob/v3.6.0/native/cocos/scene/Pass.cpp#L310 is error. It implicitly convert to Vec2. The data type may be Vec2, Vec3, Vec4 and mat4, just add a new function to convert it.

The error is trigged after i changed 
```c++
Vec2(const float *array);
```
to
```c++
explicit Vec2(const float *array);
```